libs/core/src/lib/form/form-control/form-control.directive.ts

```diff
@@ -20,36 +22,55 @@ export class FormControlDirective extends AbstractFdNgxClass {
      *  Can be `valid`, `error`, `warning` or blank for default.
      */
     @Input()
-    state: string;
+    state: FormStates;
+
+    /**
+     * Whether form is in compact mode
+     */
+    @Input()
+    compact: boolean = false;
 
 
     @Input()
     type: string;
 
     /** @hidden */
     _setProperties() {
-
-        this._addClassToElement('fd-form-control');
         if (this.state) {
             this._addClassToElement('is-' + this.state);
         }
 
         switch (this.type) {
```

Comment:
I'd recommend to split logic into separate functions like:

```
private _addControlClass(className: string, compactClassName: string): void {
     this._addClassToElement(className);
     if (this.compact) {
         this._addClassToElement(compactClassName);
         // OR this._addClassToElement(`${className}--compact`); and get rid of  compactClassName argument
    }
}
````
And then:
```
case 'checkbox': {
    this._addControlClass('fd-checkbox', 'fd-checkbox--compact');
    break;
}
```

Or create key (type/elementTag) - value (css class) map and use it instead of switch case statement.

Reply:
definitely second option because later we could get rid off of unnecessary if statements and whole switch statement can be removed

Reply:
That's great idea, actually we won't use this directive on `radio` and `checkbox` elements, which will have their own components.

Reply:
And AFAIK we don't want to depend styles on tag names on `fundamental-styles`